lib/SILGen/SILGenPoly.cpp

```diff
@@ -399,6 +399,12 @@ ManagedValue Transform::transform(ManagedValue v,
     inputOTK = OTK_None;
   }
 
+  if (inputOTK != OTK_None && outputOTK == OTK_None) {
+    SILValue result = SGF.B.createUncheckedBitCast(Loc, v.getValue(),
```

Comment:
If the input is optional and the output is not, you're forcing off a level of optionality. Maybe something is going in the wrong direction?

Also in general, `Optional<T>` is not bitwise-castable to `T` -- this only holds for classes.